checks/prometheus_check.py

```diff
@@ -342,18 +424,26 @@ def _submit(self, metric_name, message, send_histograms_buckets=True, custom_tag
         """
         if message.type < len(self.METRIC_TYPES):
             for metric in message.metric:
+                # if hostname is not specified, look at label_to_hostname setting
+                if hostname is None and self.label_to_hostname is not None:
+                    for label in metric.label:
+                        if label.name == self.label_to_hostname:
+                            custom_hostname = label.value
+                            break
+                else:
+                    custom_hostname = hostname
```

Comment:
Are you not afraid of an `UnboundLocalError` in case of the label.name is never matched ?